Render error identifier when ran verbose

SanderMuller opened the pull request:
PHPStan when ran verbose (-v) displays the [error identifier](https://phpstan.org/error-identifiers). This is very useful when solving errors or choosing to ignore this type of error, either globally or locally

### Global ignore example

In `phpstan.neon`
```neon file="phpstan.neon"
    ignoreErrors:
        - identifier: identical.alwaysFalse
          path: src/ErrorFormatter
```

### Local ignore example

```php
// @phpstan-ignore identical.alwaysFalse
if ('a' === 'b') {
    // test
}
```

Currently Simplify does not support rendering the error identifier, which is what keeps me from fully using it. This PR adds support for rendering the error identifier.

Example output:
`vendor/bin/phpstan analyse --error-format symplify -v`

```
 14/14 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100% < 1 sec

 --------------------------------------------------------------------------------------------------------
  src/ErrorFormatter/SymplifyErrorFormatter.php:143
 --------------------------------------------------------------------------------------------------------
  - '#Strict comparison using \=\=\= between 'a' and 'b' will always evaluate to false#'
  🪪 identical.alwaysFalse
 --------------------------------------------------------------------------------------------------------


                                                                                                                
 [ERROR] Found 1 errors                                                                                                                                  
```

`vendor/bin/phpstan analyse --error-format symplify` (without the `-v` verbose flag) has no changes.


For reference, the PHPStan error formatter output with the `-v` verbose flag:

`vendor/bin/phpstan analyse -v`

```
 14/14 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100% < 1 sec

 ------ -------------------------------------------------------------------------------- 
  Line   src/ErrorFormatter/SymplifyErrorFormatter.php                                   
 ------ -------------------------------------------------------------------------------- 
  143    Strict comparison using === between 'a' and 'b' will always evaluate to false.  
         🪪  identical.alwaysFalse                                                       
 ------ -------------------------------------------------------------------------------- 


                                                                                                                
 [ERROR] Found 1 error                                                                                          
```

TomasVotruba commented:
Thanks @SanderMuller , I've wanted this for a long time :clap: 

I've just approve the CI... could you check it?

SanderMuller commented:
> Thanks @SanderMuller , I've wanted this for a long time 👏
> 
> I've just approve the CI... could you check it?

Thanks! I fixed PHPStan. Seems my test broke it. Shame the base `getAnalysisResult` doesnt work since it doesn't support adding an identifier, maybe my next PR to PHPStan haha

SanderMuller commented:
> > Thanks @SanderMuller , I've wanted this for a long time 👏
> > I've just approve the CI... could you check it?
> 
> Thanks! I fixed PHPStan. Seems my test broke it. Shame the base `getAnalysisResult` doesnt work since it doesn't support adding an identifier, maybe my next PR to PHPStan haha

The 6th error in the base class actually has an identifier, so used that one now @TomasVotruba

SanderMuller commented on src/Console/Terminal.php:
```diff
@@ -127,7 +127,7 @@ private static function readFromProcess(string $command): ?string
     private static function getAnsiconWidth(): ?int
     {
-        if (!is_string(\getenv('ANSICON'))) {
+        if (! is_string(\getenv('ANSICON'))) {
```
`composer fix-cs` changed this

TomasVotruba replied:
:+1:

SanderMuller commented on src/TypeResolver/ClassConstFetchReturnTypeResolver.php:
```diff
@@ -10,7 +10,6 @@
 use PhpParser\Node\Identifier;
 use PhpParser\Node\Name;
 use PHPStan\Reflection\MethodReflection;
-use PHPStan\Reflection\ParametersAcceptorSelector;
```
`composer fix-cs` changed this

TomasVotruba replied:
:+1:

TomasVotruba commented on tests/ErrorFormatter/SymplifyErrorFormatterVerboseTest.php:
```diff
@@ -54,8 +54,10 @@ public static function getAdditionalConfigFiles(): array
 
     protected function getAnalysisResult($numFileErrors, int $numGenericErrors): AnalysisResult
     {
+        // @phpstan-ignore phpstanApi.constructor
```
We don't use docblocks to avoid missed errors.
Could you put it to phpstan.neon?

SanderMuller replied:
It's already removed in a later change

TomasVotruba commented:
> vendor/bin/phpstan analyse -v

I'm curious... does PHPStan native error formatter display identifier also only with `-v`?

TomasVotruba commented:
Re-approved, these is new CI feedback :+1:

SanderMuller commented:
> > vendor/bin/phpstan analyse -v
> 
> I'm curious... does PHPStan native error formatter display identifier also only with `-v`?

Yes

SanderMuller commented:
> Re-approved, these is new CI feedback :+1: 

Tests were passing on Windows, will fix it from my Mac in a bit. 😅

TomasVotruba commented:
@SanderMuller Don't worry about testing this, it's pretty straighforward :)

I've run classic PHPStan and display identifier out of the box, even without `-v`. I think this one should have the same behavior:

![Screenshot From 2025-01-07 08-08-01](https://github.com/user-attachments/assets/fdc45a6b-9f59-4936-b056-e435f3baf9c4)

SanderMuller commented:
> @SanderMuller Don't worry about testing this, it's pretty straighforward :)
> 
> I've run classic PHPStan and it displays the identifier out of the box, even without `-v`. I think this one should have the same behavior:
> 
> ![Screenshot From 2025-01-07 08-08-01](https://private-user-images.githubusercontent.com/924196/400686166-fdc45a6b-9f59-4936-b056-e435f3baf9c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyMzgwMDUsIm5iZiI6MTczNjIzNzcwNSwicGF0aCI6Ii85MjQxOTYvNDAwNjg2MTY2LWZkYzQ1YTZiLTlmNTktNDkzNi1iMDU2LWU0MzVmM2JhZjljNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwN1QwODE1MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xM2FjYjI2MzNhMTk5MmJmOWY0NWI4NmU5ZGI1ZWFhYjVlZjYxMTkxMmY1ZGJlNDVkNzA2MzZkMzhmNDQwMzE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.I5irVaHuIRz6-tc3aeSxhdhdDoYsn3LuDZDk4Nvl5lk)

I guess PHPStan 2.0 added this by default instead of opt-in. I think identifiers became mandatory in v2 and therefor making sense that it's rendered by default.
Updated this PR to render the identifier by default instead of only when verbose and removed the verbose tests

TomasVotruba commented:
Awesome, let's ship this :pray:

TomasVotruba commented:
Thank you :+1:

TomasVotruba commented:
Just tagged... give it a try: https://github.com/symplify/phpstan-extensions/releases/tag/12.0.1

SanderMuller commented:
> Just tagged... give it a try: https://github.com/symplify/phpstan-extensions/releases/tag/12.0.1

Yay, it works how I want it to work! Thanks for the quick review, merge & tag!
